fastly/resource_fastly_service_wasm_v1_test.go

```diff
@@ -0,0 +1,88 @@
+package fastly
+
+import (
+	"fmt"
+	"testing"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccFastlyServiceWASMV1_basic(t *testing.T) {
+	var service gofastly.ServiceDetail
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	domainName1 := fmt.Sprintf("fastly-test1.tf-%s.com", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceWASMV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceWASMV1Config(name, domainName1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_wasm.foo", &service),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "comment", "Managed by Terraform"),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "version_comment", ""),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "active_version", "0"),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "domain.#", "1"),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "backend.#", "1"),
+					resource.TestCheckResourceAttr(
+						"fastly_service_wasm.foo", "package.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckServiceWASMV1Destroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "fastly_service_wasm" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		l, err := conn.ListServices(&gofastly.ListServicesInput{})
+		if err != nil {
+			return fmt.Errorf("[WARN] Error listing servcies when deleting Fastly Service (%s): %s", rs.Primary.ID, err)
+		}
+
+		for _, s := range l {
+			if s.ID == rs.Primary.ID {
+				// service still found
+				return fmt.Errorf("[WARN] Tried deleting Service (%s), but was still found", rs.Primary.ID)
+			}
+		}
+	}
+	return nil
+}
+
```

Comment:
Would you mind adding a test for an empty `package` block (if this is even possible)? That way a reader can easily see the expected behavior?

Reply:
Filename is now required